api_tests/applications/views/test_application_detail.py

```diff
@@ -86,131 +104,134 @@ def setUp(self):
+                                     'home_url': user_app_copy.home_url,
+                                     'callback_url': user_app_copy.callback_url
                                      },
                                     res.json['data']['attributes'])
 
-    def test_updating_an_instance_does_not_change_the_number_of_instances(self):
-        new_name = "The instance formerly known as Prince"
-        res = self.app.patch_json_api(self.user1_app_url,
+    def test_updating_an_instance_does_not_change_the_number_of_instances(self, app, user, user_app, user_app_url):
+        new_name = 'The instance formerly known as Prince'
+        res = app.patch_json_api(user_app_url,
                                       {'data': {
-                                          'attributes': {"name": new_name},
-                                          'id': self.user1_app.client_id,
-                                          'type': 'applications'}}, auth=self.user1.auth)
-        assert_equal(res.status_code, 200)
+                                          'attributes': {'name': new_name},
+                                          'id': user_app.client_id,
+                                          'type': 'applications'}}, auth=user.auth)
+        assert res.status_code == 200
 
         list_url = _get_application_list_url()
-        res = self.app.get(list_url, auth=self.user1.auth)
-        assert_equal(res.status_code, 200)
-        assert_equal(len(res.json['data']),
-                     1)
+        res = app.get(list_url, auth=user.auth)
+        assert res.status_code == 200
+        assert len(res.json['data']) == 1
 
     @mock.patch('framework.auth.cas.CasClient.revoke_application_tokens')
-    def test_deleting_application_flags_instance_inactive(self, mock_method):
+    def test_deleting_application_flags_instance_inactive(self, mock_method, app, user, user_app, user_app_url):
         mock_method.return_value(True)
-        res = self.app.delete(self.user1_app_url, auth=self.user1.auth)
-        self.user1_app.reload()
-        assert_false(self.user1_app.is_active)
+        res = app.delete(user_app_url, auth=user.auth)
+        user_app.reload()
+        assert not user_app.is_active
 
-    def test_update_application(self):
-        res = self.app.put_json_api(self.user1_app_url, self.correct, auth=self.user1.auth, expect_errors=True)
-        assert_equal(res.status_code, 200)
+    def test_update_application(self, app, user, user_app, user_app_url, missing_id, missing_type, incorrect_id, incorrect_type, correct):
+        res = app.put_json_api(user_app_url, correct, auth=user.auth, expect_errors=True)
+        assert res.status_code == 200
 
-    def test_update_application_incorrect_type(self):
-        res = self.app.put_json_api(self.user1_app_url, self.incorrect_type, auth=self.user1.auth, expect_errors=True)
-        assert_equal(res.status_code, 409)
+    #   test_update_application_incorrect_type
```

Comment:
Were these tests combined to make them faster? In general, it's better to have focused tests that test one thing. Do we get a lot of time savings if we combine these?